test/field_types/test_list_field.py

```diff
@@ -33,3 +38,24 @@ def test_validate(self):
 
     def test_get_default(self):
         self.assertEqual([], self.field.get_default())
+
+    def test_mutable_default(self):
+        article = self.Article()
+        self.assertIs(article.tags, article.tags)
+
+        article.tags.append('foo')
+        article.tags.append('bar')
+        self.assertEqual(article.tags, ['foo', 'bar'])
+
+        # Ensure tags is saved to the database.
+        article.save()
+        self.assertEqual(article.tags, self.Article.objects.first().tags)
+        self.assertEqual(article.tags, ['foo', 'bar'])
+
+        # Ensure that deleting a field restores the default value.
+        del article.tags
+        self.assertEqual(article.tags, [])
+
+        # Ensure the deleted field is removed from the database.
+        article.save()
+        self.assertNotIn('tags', DB.article.find_one())
```

Comment:
This is tricky. 
```python

        # Ensure that a refresh restores the default value.
        article.tags.append('foo')
        self.assertEqual(article.tags, ['foo'])
        article.refresh_from_db()
        self.assertEqual(article.tags, [])
```

For the above to pass you'll need to clear the defaults in MongoModelBase._set_attributes

```python
    def _set_attributes(self, dict):
        """Set this object's attributes from a dict."""
        self._data.clear()
        self._defaults.clear()
```